### ISSUE TITLE
fail to search for index in directory with directory name containing CJK characters

### ISSUE THREAD

**Issue by wh-timme** [sanic-org/sanic#3008]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

file path:
./public/hello/你好/index.html
will return 404 when accessing to http://serving_domain/hello/你好

### Code snippet

app.static('/', './public', index='index.html')

### Expected Behavior

redirect to ./public/hello/你好/index.html instead of 404 page

### How do you run Sanic?

As a script (`app.run` or `Sanic.serve`)

### Operating System

Linux

### Sanic Version

24.6.0

**Comment by adk23333** [comment]
Anyway, you shouldn't use chinese path.
不建议使用中文目录，这会出现什么问题，我也不知道。

**Comment by Tronic** [comment]
How are CJK pathnames handled nowadays? Unicode or some other weirdness? To my knowledge, Sanic and Python ought to handle Unicode conversions properly, if the browser supplies the path in UTF-8.

To triage, could you print `request.url_bytes` and `request.url`, and try if you can open the file with Python open(...), and if so, how do you need to write the CJK characters in that.

**Comment by adk23333** [comment]
> How are CJK pathnames handled nowadays? Unicode or some other weirdness? To my knowledge, Sanic and Python ought to handle Unicode conversions properly, if the browser supplies the path in UTF-8.

It is usually URL encoding. 
[example website](https://www.urlencoder.io/)

**Comment by Tronic** [comment]
Yes, but that doesn't tell which encoding. UTF-8 and other encodings produce different %-codes for the same character, and I have no idea what you actually use in China.

**Comment by adk23333** [comment]
> UTF-8 and other encodings produce different %-codes for the same character, and I have no idea what you actually use in China.


Most people generally use UTF - 8 encoding, and some old websites may use GBK.

example as 

https://fanyi.baidu.com/#en/zh/Yes%2C%20but%20that%20doesn't%20tell%20which%20encoding.%20UTF-8%20and%20other%20encodings%20produce%20different%20%25-codes%20for%20the%20same%20character%2C%20and%20I%20have%20no%20idea%20what%20you%20actually%20use%20in%20China.
